flocker/control/_protocol.py

```diff
@@ -253,7 +315,19 @@ def __init__(self, agent):
         CommandLocator.__init__(self)
         self.agent = agent
 
+    @property
+    def logger(self):
+        """
+        ``with_eliot_context`` assumes that the decorated method's bound
+        instance has a logger, which it could, but for testing, it needs to use
+        the same logger as the supplied agent.
+
+        XXX This seems ugly.
+        """
+        return self.agent.logger
```

Comment:
`IAgent` doesn't have a `logger` attribute.  I see how this can work in the test suite, since `FakeAgent` has been given this attribute - but I don't see where it will come from in production when using an actual agent from Twisted.